modcc/visitor.hpp

```diff
@@ -59,3 +61,119 @@ class Visitor {
 
     virtual ~Visitor() {};
 };
+
+// Visitor specialization intended for use as a base class for visitors that
+// operate as function or procedure body rewriters after semantic analysis.
+//
```

Comment:
can you please expand on this comment to explain what rewriting involves/is? I ask this knowing full well that most of my original contributions were sparsely documented.